resources/js/Pages/Galleries/hooks/useWalletDraftGalleries.ts

```diff
@@ -0,0 +1,218 @@
+    }, [address]);
+
+    useEffect(() => {
+        void updateDraftState();
+    }, [address]);
+
+    /**
+     * Add new draft gallery.
+     *
+     * @param {GalleryDraft} draft
+     * @returns {Promise<GalleryDraft>}
+     */
+    const add = async (draft: GalleryDraft): Promise<GalleryDraft> => {
+        const { id: _, ...draftToSave } = draft;
+        const drafts = await allDrafts();
+
+        if (drafts.length >= MAX_DRAFT_LIMIT_PER_WALLET) {
+            throw new Error("[useWalletDraftGalleries:add] Limit Reached");
+        }
+
+        setIsSaving(true);
+
+        const id = await database.add({
+            ...draftToSave,
+            updatedAt: new Date().getTime(),
+        });
+
+        setIsSaving(false);
+
+        return await findByIdOrThrow(id);
+    };
+
+    /**
+     * Update existing gallery.
+     *
+     * @param {GalleryDraft} draft
+     * @returns {Promise<GalleryDraft>}
+     */
+    const update = async (draft: GalleryDraft): Promise<GalleryDraft> => {
+        if (!isTruthy(draft.id)) {
+            throw new Error("[useWalletDraftGalleries:update] Missing Id");
+        }
+
+        setIsSaving(true);
+
+        await database.update({
+            ...draft,
+            updatedAt: new Date().getTime(),
+        });
+
+        setIsSaving(false);
+
+        return await findByIdOrThrow(draft.id);
+    };
+
+    /**
+     * Upsert gallery.
+     *
+     * @param {GalleryDraft} draft
+     * @returns {Promise<GalleryDraft>}
+     */
+    const upsert = async (draft: GalleryDraft): Promise<GalleryDraft> => {
+        if (isTruthy(draft.id)) {
+            return await update(draft);
+        }
+
+        return await add(draft);
+    };
+
+    /**
+     * Remove gallery.
+     *
+     * @param {number} id
+     * @returns {Promise<void>}
+     */
+    const remove = async (id?: number | null): Promise<void> => {
+        if (isTruthy(id)) {
+            await database.deleteRecord(id);
+            await updateDraftState();
+        }
+    };
+
+    /**
+     * Remove expired galleries.
+     *
+     * @returns {Promise<void>}
+     */
+    const removeExpired = async (): Promise<void> => {
+        const drafts: GalleryDraft[] = await database.getAll();
+
+        for (const draft of drafts) {
+            if (isExpired(draft) && isTruthy(draft.id)) {
+                void remove(draft.id);
+            }
+        }
+    };
+
+    /**
+     * Get all drafts.
+     *
+     * @returns {Promise<GalleryDraft[]>}
+     */
+    const allDrafts = async (): Promise<GalleryDraft[]> => {
+        const allDrafts: GalleryDraft[] = await database.getAll();
+        return allDrafts.filter((draft) => draft.walletAddress === address && !isExpired(draft));
+    };
+
+    /**
+     * Find gallery.
+     *
+     * @param {number} id
+     * @returns {Promise<GalleryDraft>}
+     */
+    const findById = async (id: number | string): Promise<GalleryDraft | undefined> => {
```

Comment:
Would you consider naming it as `findWalletDraftById` because of the address check?